src/overlays/actors/ovl_Item_Shield/z_item_shield.c

```diff
@@ -138,38 +147,51 @@ void func_80B86BC8(ItemShield* this, PlayState* play) {
     }
 }
 
-void func_80B86CA8(ItemShield* this, PlayState* play) {
-    static Vec3f D_80B871F4 = { 0.0f, 0.0f, 0.0f };
-    static f32 D_80B87200[] = { 0.3f, 0.6f,  0.9f, 1.0f,  1.0f, 1.0f,  1.0f, 1.0f,
-                                1.0f, 0.85f, 0.7f, 0.55f, 0.4f, 0.25f, 0.1f, 0.0f };
-    static f32 D_80B87240[] = { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.8f,
-                                0.6f, 0.4f, 0.2f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+#define ITEMSHIELD_FLAME_TYPE_COUNT 15
+
+void ItemShield_Burning_Combust(ItemShield* this, PlayState* play) {
+    static Vec3f sFlameSpawnPos = { 0.0f, 0.0f, 0.0f };
+    static f32 sFlameScales[] = {
+        0.3f, 0.6f, 0.9f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.85f, 0.7f, 0.55f, 0.4f, 0.25f, 0.1f, 0.0f,
+    };
+    static f32 sFlameIntensities[] = {
+        1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.8f, 0.6f, 0.4f, 0.2f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+    };
     s32 i;
-    s32 temp;
+    s32 type;
 
     Actor_MoveXZGravity(&this->actor);
     Actor_UpdateBgCheckInfo(play, &this->actor, 10.0f, 10.0f, 0.0f, UPDBGCHECKINFO_FLAG_0 | UPDBGCHECKINFO_FLAG_2);
     this->actor.shape.yOffset = ABS(Math_SinS(this->actor.shape.rot.x)) * 1500.0f;
 
-    for (i = 0; i < 8; i++) {
-        temp = 15 - this->unk_19E[i];
-        D_80B871F4.x = this->unk_1A8[i].x;
-        D_80B871F4.y = this->unk_1A8[i].y + (this->actor.shape.yOffset * 0.01f) + (D_80B87200[temp] * -10.0f * 0.2f);
-        D_80B871F4.z = this->unk_1A8[i].z;
-        EffectSsFireTail_SpawnFlame(play, &this->actor, &D_80B871F4, D_80B87200[temp] * 0.2f, -1, D_80B87240[temp]);
-        if (this->unk_19E[i] != 0) {
-            this->unk_19E[i]--;
+    // Spawn flames at random locations until despawn
+    for (i = 0; i < ITEMSHIELD_FLAME_COUNT; i++) {
+        type = ITEMSHIELD_FLAME_TYPE_COUNT - this->flameTypes[i];
+
+        sFlameSpawnPos.x = this->flamePos[i].x;
+        sFlameSpawnPos.y =
+            this->flamePos[i].y + (this->actor.shape.yOffset * 0.01f) + (sFlameScales[type] * -10.0f * 0.2f);
+        sFlameSpawnPos.z = this->flamePos[i].z;
+
+        EffectSsFireTail_SpawnFlame(play, &this->actor, &sFlameSpawnPos, sFlameScales[type] * 0.2f, -1,
+                                    sFlameIntensities[type]);
+
+        // cycle through flame types and pick new random positions until the shield is about to disappear
+        if (this->flameTypes[i] != 0) {
+            this->flameTypes[i]--;
         } else if (this->timer > 16) {
-            this->unk_19E[i] = 15;
-            this->unk_1A8[i].x = Rand_CenteredFloat(15.0f);
-            this->unk_1A8[i].y = Rand_CenteredFloat(10.0f);
-            this->unk_1A8[i].z = Rand_CenteredFloat(15.0f);
+            this->flameTypes[i] = ITEMSHIELD_FLAME_TYPE_COUNT;
```

Comment:
This means `ITEMSHIELD_FLAME_TYPE_COUNT` is a flame type
and all values down to 0 are flame types
so there would be `ITEMSHIELD_FLAME_TYPE_COUNT+1` flame type values
i.e. `ITEMSHIELD_FLAME_TYPE_COUNT` is misnamed, idk how to fix but it could be 16 instead and use `ITEMSHIELD_FLAME_TYPE_COUNT-1` here

also then `sFlameScales[ITEMSHIELD_FLAME_TYPE_COUNT]` and `sFlameIntensities[ITEMSHIELD_FLAME_TYPE_COUNT]`

Reply:
Had a quick look at what's going on. 

this->flameTypes essentially stores the remaining "life" for 8 different flames. It starts by initializing the 8 flames to the odds (1, 3, ..., 15) based on the number of flames. Then, each call to `ItemShield_Burning_Combust` draws the flame by computing the current animation frame (`type = 15 -  this->flameTypes`), and then calling EffectSsFireTail_SpawnFlame. Finally, if the "life" count is greater than zero, it gets decremented, otherwise a new flame is "spawned" with a life of 15 as long as there is enough time remaining.

So I think `this->flameTypes` -> `this->flameLife`, `type` -> `animFrame`, `ITEMSHIELD_FLAME_TYPE_COUNT` -> ITEMSHIELD_FLAME_LIFE_MAX` might be the way to go here, and I think a comment at line 82 is warranted to point out the weird coupling of `ITEMSHIELD_FLAME_COUNT` to the flame life.